shards/shards.go

```diff
@@ -497,6 +499,9 @@ func (ss *shardedSearcher) streamSearch(ctx context.Context, proc *process, q qu
 
 	defer cancel()
 
+	mu := sync.Mutex{}
+	pendingPriorities := prioritySlice{}
```

Comment:
Since this type is always used concurrently, would it make sense to make it concurrent safe internally by making it a struct and embedding the lock there?

Reply:
@tsenart the lock is held while we send. It is documented below that we rely on the stream to have a monotonically decreasing value for `MaxPendingShardPriority`. If we released the lock after updating `pendingPriorities`, it is possible it could race with another send and we would violate that property.

Note: the docs don't use the words "monotonically decreasing", maybe that would be useful way to document it so we can re-derive the need for the lock.